pwnlib/rop/rop.py

```diff
@@ -1277,6 +1277,24 @@ def search(self, move = 0, regs = None, order = 'size'):
 
         return result
 
+    def ret2dlresolve(self, reloc_index, real_args, read_func, read_func_args):
+        elf = [elf for elf in self.elfs if elf.get_section_by_name(".plt") is not None][0]
+        elf_base = elf.address if elf.pie else 0
+        plt_init = elf.get_section_by_name(".plt").header.sh_addr + elf_base
+        log.debug("PLT_INIT: %s", hex(plt_init))
+
+        self.call(read_func, read_func_args)
```

Comment:
I think getting `read_func` and `read_func_args` can be somewhat automated having the elf, but I'm not sure how.